### PR TITLE
Loading spinner in modules

### PR DESCRIPTION
**Pull request by agudek**
Multi-color spinners are added before module body is added to the svg and hidden after.
Go to index.html#tab-project to see it in action in the 'Number of pull-request per number of comments' module

### CODE REVIEW

**Comment by mboom** [comment]
It works! Nice feature!

File: src/index.html [hunk 1]
```diff
@@ -4,6 +4,53 @@
...
+                        </div><div class="gap-patch">
+                          <div class="circle"></div>
+                        </div><div class="circle-clipper right">
+                          <div class="circle"></div>
+                        </div>
+                      </div>
+                    </div>
+                </div>
+             </script>
+
```
Comment (DaanvanderValk) [inline src/index.html]: Should this be moved to an external script file?

Comment (agudek) [inline src/index.html]: Will look if that is possible

File: src/main.js [hunk 2]
```diff
@@ -4,130 +4,132 @@
...
-                        Axis.tickSize(720-50-50);
-                    }
+    function scaleAxis(module, objects, axisname) {
+        /*jshint maxcomplexity:7 */
+        if(octopeerHelper.getSafeModuleValue(module,axisname+"AxisScale")() === "fit"){
+            var Scale = d3.scale.linear()
+                .domain(octopeerHelper.getSafeModuleValue(module,axisname+"AxisFitFunction")(objects))
+                .range([350-50-10,0])
+                .nice();
+            var Axis = d3.svg.axis().scale(Scale);
```
Comment (borek2) [inline src/main.js]: Capital letters ;)

File: src/main.js [hunk 3]
```diff
@@ -4,130 +4,132 @@
...
-                    }
-                } else {
-                    Axis.orient("right");
-                    if(octopeerHelper.getSafeModuleValue(module,axisname+"AxisTicks")) {
-                        Axis.tickSize(720-50-50);
-                    }
+    function scaleAxis(module, objects, axisname) {
+        /*jshint maxcomplexity:7 */
+        if(octopeerHelper.getSafeModuleValue(module,axisname+"AxisScale")() === "fit"){
+            var Scale = d3.scale.linear()
```
Comment (borek2) [inline src/main.js]: Capital letters ;)

Comment (agudek) [inline src/main.js]: @borek2 Those have been fixed in another branch, which is probably the reason for the merge conflicts. Will resolve them now.

**Pull request merged**